transport/server.cc

```diff
@@ -577,6 +581,13 @@ future<> cql_server::connection::process_request() {
         }
 
         auto& f = *maybe_frame;
+
+        const bool allow_shedding = _client_state.get_workload_type() == service::client_state::workload_type::interactive;
+        if (allow_shedding && _shed_incoming_requests) {
+            ++_server._stats.requests_shed;
+            return _read_buf.skip(f.length);
+        }
```

Comment:
This is a little scary. We could spend many cycles looping and shedding requests, while those cycles can be used to retire existing requests. But the 50ms timer offers some protection, and serving a request is much more expensive than shedding. I think it's okay.